registry/registry.go

```diff
@@ -219,6 +220,10 @@ func ContinueOnError(err error) bool {
 		return shouldV2Fallback(v)
 	case *client.UnexpectedHTTPResponseError:
 		return true
+	case error:
+		if val := strings.Contains(err.Error(), errNoSpaceOnDevice.Error()); val {
```

Comment:
I'm not sure what the real error is that is returned here (maybe you can provide some insight on the actual error type here since you've been working on it?)... but we should compare this error to `syscall.ENOSPC` rather than defining `errNoSpaceOndevice`